### DISCUSSION TITLE
Where are bookmarks/notes stored?

### DISCUSSION THREAD

**Discussion by MSchabel** [devnoname120/epubviewer#25] [Q&A] [answered]
As the title says, I was not able to figure it out myself. Are they stored directly in the Epub file? Or somewhere else on the nextcloud server? Would like to be able to back them up somehow.

**Comment by devnoname120** [comment, marked as answer]
Everything is stored in the [Nextcloud database](https://docs.nextcloud.com/server/latest/admin_manual/configuration_database/linux_database_configuration.html):
- Bookmarks and notes are stored in the table `reader_bookmarks`.
- Preferences in `reader_prefs`.
